psydac/linalg/tests/test_block.py

```diff
@@ -989,6 +989,34 @@ def test_block_linear_operator_parallel_dot( dtype, n1, n2, p1, p2, P1, P2 ):
     # Check data in 1D array
     assert np.allclose( Y.blocks[0].toarray(), y1.toarray(), rtol=1e-14, atol=1e-14 )
     assert np.allclose( Y.blocks[1].toarray(), y2.toarray(), rtol=1e-14, atol=1e-14 )
+
+    # Test copy with an out 
+    K = BlockLinearOperator( W, W )
+    L.copy(out=K)
+
+    # Compute Block-vector product
+    K.dot(X, out= Y)
+
+    # Check data in 1D array
+    assert np.allclose( Y.blocks[0].toarray(), y1.toarray(), rtol=1e-14, atol=1e-14 )
+    assert np.allclose( Y.blocks[1].toarray(), y2.toarray(), rtol=1e-14, atol=1e-14 )
+
+    # Test transpose with an out
+    N = BlockLinearOperator( W, W )
+    L.transpose(out = N)
```

Comment:
Same comment. Maybe we need to write into `N` (and `K`) at least twice in order to properly test the `out` argument?

Reply:
Is what I did in [71f2d07](https://github.com/pyccel/psydac/pull/357/commits/71f2d07fad81f77597e52e417c8707c1b93661a7) what you expected?

Reply:
Yes! The test is much stronger now!